### PR TITLE
Guardduty construct

### PR DESCRIPTION
**Pull request by parkand1**
Amazon GuardDuty and SNS pattern for EKS Blueprints

### CODE REVIEW

File: lib/guardduty-construct/index.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import { aws_guardduty as guardduty, aws_events as events,
```
Comment (shapirov103) [inline lib/guardduty-construct/index.ts:1]: minor: Number of items per import is 3. Then separate line and wildcard, e.g.
 
 ```
import * as guardduty from aws-cdk-lib/aws_guardduty;
 ```

File: lib/guardduty-construct/index.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    // The email address to send security findings to
+    email: string;
+}
+const defaultProps: GuardDutyNotifierProps = {
+    environmentName: "main",
+    email: 'your-email',
+};
+
+export default class GuardDutyNotifier extends Stack {
+    constructor(scope: Construct, id: string, _props?: StackProps){
```
Comment (shapirov103) [inline lib/guardduty-construct/index.ts:19]: There has to be a rule in ESLint to ignore variables with 
"_". I see github action warning now for the unused variable.

**Review by shapirov103 — COMMENTED** [review]
Let's wait for Patrol features (and RDS) to be included or at least clarity on when those will be part of CDK.

File: lib/guardduty-construct/index.ts (hunk 1 above)
Comment (parkand1) [reply lib/guardduty-construct/index.ts]: Addressed

File: lib/guardduty-construct/index.ts (hunk 2 above)
Comment (parkand1) [reply lib/guardduty-construct/index.ts]: This was addressed after merging Aliakseis PR

File: lib/security/guardduty-construct/index.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  async buildAsync(scope: Construct, id: string) {
+
+    await prevalidateSecrets(process.env.CDK_DEFAULT_REGION!, SECRET_ARGO_ADMIN_PWD);
+
+    const stackID = `${id}-blueprint`;
+
+
+    blueprints.EksBlueprint.builder()
+      .account(process.env.CDK_ACCOUNT_ID!)
```
Comment (youngjeong46) [inline lib/security/guardduty-construct/index.ts:24]: Isn't this `CDK_DEFAULT_ID`?

Comment (shapirov103) [reply lib/security/guardduty-construct/index.ts:24]: +1 please address

File: lib/security/guardduty-construct/guard-duty-setup.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  ): blueprints.NestedStackBuilder {
+    return {
+      build(scope: Construct, id: string, props: NestedStackProps) {
+        return new GuardDutySetupStack(
+          scope,
+          id,
+          props,
+          environmentName,
+          email,
+          dataSources
```
Comment (shapirov103) [inline lib/security/guardduty-construct/guard-duty-setup.ts:27]: Let's fix to use features instead of datasources as those are deprecated.

**Review by shapirov103 — COMMENTED** [review]
a couple of tactical comments

File: lib/security/guardduty-construct/index.ts (hunk 3 above)
Comment (parkand1) [reply lib/security/guardduty-construct/index.ts:19]: addressed

Comment (shapirov103) [reply lib/security/guardduty-construct/index.ts:19]: @parkand1 not addressed? should be default account. How did it work in tests? Did you set that particular env variable in your bash/zsh profile?

**Review by shapirov103 — COMMENTED** [review]
@parkand1 please see one comment below. Also confirm the test succeeded.

File: lib/security/guardduty-construct/guard-duty-setup.ts (hunk 4 above)
Comment (parkand1) [reply lib/security/guardduty-construct/guard-duty-setup.ts]: addressed

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**